src/cli/rustup_mode.rs

```diff
@@ -808,9 +811,43 @@ fn update(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
         common::update_all_channels(cfg, self_update, m.is_present("force"))?;
     }
 
+    delete_downloads_dir_contents();
+
     Ok(())
 }
 
+fn delete_downloads_dir_contents() {
+    match dirs::home_dir() {
+        Some(path) => match path.to_str() {
+            Some(home_dir) => {
+                let mut downloads_dir_path = "".to_string();
+                downloads_dir_path.push_str(&home_dir);
+                downloads_dir_path.push_str(&"/.rustup/downloads".to_string());
```

Comment:
Couldn't this just be `downloads_dir_path.push_str("/.rustup/downloads");` ?

Reply:
OR, even simpler: 
`let downloads_dir_path = format!("{}/.rustup/downloads", home_dir);`